Handling form validation, fix some i18n bugs and add missing translations

ph1p opened the pull request:
Buttercup shows error popups on saving empty meta data objects (custom fields). This PR should fix it.
Maybe this is a part of buttercup-core.

**Reproduce**
- Create or edit entry
- add empty custom field(s)
- click save

**Issues**: #363, #371

perry-mitchell commented:
Thanks for the PR @ph1p - Unfortunately I don't know if this is the best way to accomplish this fix. It's absolutely a problem, but forcibly filtering out "invalid" or empty meta values is perhaps a bit heavy handed and dangerous. It would be best to have some kind of **form validation** that is executed when clicking save. Errors could be pointed out and saving prevented, if there are issues.

ph1p commented:
You're right it's a little bit quick an dirty. I'm already on it!
I'm renaming this PR to "Handling Form Validation".

perry-mitchell commented:
@ph1p Remember to not add to much to the same PR.. Every pull request should target perhaps 1 issue and not try to do too much else. I wouldn't personally add any more translation content because we have to re-verify all of that.

Keep up the awesome work! ☺

ph1p commented:
My mistake. While I was coding, I found these errors/untranslated words and fixed them immediately. Next time I'll write them down and split them into different PRs. This PR is ready so far or should I split it now and create new PRs?

perry-mitchell commented on src/renderer/components/archive/entry-view.js:
```diff
@@ -29,14 +29,19 @@ const EntryView = ({ entry, intl }) => (
-        ))}
+        {entry.meta.map(
+          meta =>
+            Object.keys(meta).length ? (
```
You could use jsx control statements here:

```javascript
<If condition={Object.keys(meta).length > 0}>
  <!-- content -->
</If>
```

perry-mitchell replied:
This might be more tidy than the ternary operator.

ph1p replied:
Oh cool, didn't know that :)

perry-mitchell replied:
Yeah it's great! Much nicer imo to read just XML..

perry-mitchell commented on src/renderer/components/tree-view/tree-label-edit.js:
```diff
@@ -76,3 +82,12 @@ export default class LabelEditor extends Component {
 }
+
+LabelEditor.propTypes = {
+  node: PropTypes.object,
```
Node is used without checks so should be `isRequired` here.

perry-mitchell replied:
I know that you didn't invent the code of course, but if we're changing it then it's a good time to update it 😄

ph1p replied:
I'll change it ;) Don't know if it's better to write `propTypes` in- or outside class. What do you prefer?

perry-mitchell replied:
I think `static propTypes` within classes is what we're going for now - last I spoke with @sallar :)

perry-mitchell commented:
Nice @ph1p! I'll still let @sallar check the code as this is quite a large update.

Thanks..

ph1p commented:
Ok :)

sallar commented:
Great job @ph1p
